semgrep-core/matching/Generic_vs_generic.ml

```diff
@@ -995,7 +996,17 @@ and m_list__m_argument (xsa: A.argument list) (xsb: A.argument list) =
   (* dots: ..., can also match no argument *)
   | [A.Arg (A.Ellipsis _i)], [] ->
       return ()
-
+  (* Fix to issue 1173. Case where there are ellipsis to the left and right of a statement.
+    * Order should not matter in this case, as we can see from the go structs example.
+    * However, there is an issue where this will match a correct line twice occasionally.
+    * Look into m_list_in_any_order for correction in this case. *)
+   | (A.Arg (A.Ellipsis _i))::xsa, xsb
```

Comment:
To what extent is our desire to match in any order due to the pattern syntax; vs. the parent AST node?

I might imagine instead defining two different arg matching paths, using one from, e.g., function argument list, and using the other from object, import, and struct definition lists. cc. @aryx for thoughts here.

Reply:
I'm not sure I understand what you mean, but when we are in a object/records/hash definition context, then
the semantic of the language says that the order of the field does not really matter, which is why we want to match those in any order.
For function arguments, we match certain arguments in any order, but only for the keyword arguments (ArgKwd in AST_generic.ml).

Reply:
Yup, what you say is what I was suggesting: whether a list of arguments is order-independent should probably depend on the semantic context of the language (which ideally is represented in the parent node type of the generic AST).

The change here, in contrast, I believe uses the pattern API to determine whether matches are order-independent, rather than the semantic context.

Reply:
done!